Sources/Controllers/Map/Layers/OAOverlayMapLayer.mm

```diff
@@ -141,4 +145,13 @@ - (void) updateOverlayLayer
     });
 }
 
+
+- (void)showOpacitySliderIfItEnabled
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        BOOL isSliderEnabled = [OAAppSettings sharedManager].mapSettingShowOverlayOpacitySlider;
+        [[OARootViewController instance].mapPanel updateOverlayUnderlayView:isSliderEnabled];
```

Comment:
Method updateOverlayUnderlayView should be without parameter.

Reply:
I'm not sure I understand you correctly. How can I call a method without sending a parameter? The compiler will show me an error because this parameter is required. Or am I missing something? Can you explain me, please?

https://github.com/osmandapp/OsmAnd-iOS/blob/c0ab9ad33fe2e8ef3de47ea0637ee16d694ee1ca/Sources/Controllers/Panels/OAMapPanelViewController.mm#L366

Reply:
I suggest to remove this param and read visibility from settings (since mapSettingShowOverlayOpacitySlider and mapSettingShowUnderlayOpacitySlider were introduced) in updateOverlayUnderlayView method. Actually we have bug here - if underlay is visible and we turn off overlay - both sliders will disappear from screen, but underlay slider should stay in this case.

Reply:
I fixed that bug. For that, I moved the code a little deeper. To the OAMapHudViewController and OAOverlayUnderlayView classes. As a result, the parameter in the updateOverlayUnderlayView method is no longer needed.